src/com/hms_networks/americas/sc/string/StringUtils.java

```diff
@@ -45,4 +48,35 @@ public static String replace(String string, String target, String replacement) {
     // Return modified string
     return stringBuffer.toString();
   }
+
+  /**
+   * Method to split the supplied string into smaller strings using the supplied regex as the
+   * delimiter.
+   *
+   * @param string string to split
+   * @param regex string split delimiter
+   * @return list of string split parts
+   * @since 1.2
+   */
+  public static List split(String string, String regex) {
```

Comment:
I am confused why you are calling this a regex. I see that you are trying to match what the native java function does but this naming and commenting is VERY misleading to people using this library.

https://docs.oracle.com/javase/7/docs/api/java/lang/String.html#indexOf(java.lang.String)

Reply:
I did it to match the actual Java implementation. Do you have a suggestion for a different name?

Reply:
If you wanted to match the actual java implementation you would need to handle the regex correctly.

Otherwise "delimiter" would make sense

Reply:
changed